.github/workflows/devcontainer-feature-release.yml

```diff
@@ -0,0 +1,106 @@
+name: Build dev container feature image
+
+env:
+  IMAGE_NAME: ${{ github.repository }}
+  REGISTRY: ghcr.io
+
+on:
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      matrix:
+        baseImage:
+          [
+            "ubuntu:focal",
+            "ubuntu:jammy",
+            "ubuntu:bionic",
+            "debian:11",
+            "debian:10",
+            "mcr.microsoft.com/devcontainers/base:ubuntu",
+            "mcr.microsoft.com/devcontainers/base:debian",
+          ]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: "Install latest devcontainer CLI"
+        run: npm install -g @devcontainers/cli
+
+      - name: "Generating tests for 'azd' against '${{ matrix.baseImage }}'"
+        run: devcontainer features test -f azd -i ${{ matrix.baseImage }} .
+        working-directory: ./ext/devcontainer
+
+  test-scenarios:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: "Install latest devcontainer CLI"
+        run: npm install -g @devcontainers/cli
+
+      - name: "Testing 'azd' scenarios"
+        run: devcontainer features test -f azd --skip-autogenerated .
+        working-directory: ./ext/devcontainer
+
+  test-global:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: "Install latest devcontainer CLI"
+        run: npm install -g @devcontainers/cli
+
+      - name: "Testing global scenarios"
+        run: devcontainer features test --global-scenarios-only .
+        working-directory: ./ext/devcontainer
+
+  deploy:
+    if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
Would we ever want to run these test pipelines in `main` without releasing? If so, [workflow_dispatch inputs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs) could help here.